Guide for input/output in original FW. (#20141)
openvinotoolkit/openvino: docs/articles_en/documentation/openvino_legacy_features/mo_ovc_transition.md

            - Not available in OVC tool. Please check Python API.

Cutting Off Parts of a Model
############################

> In new conversion API input and output model cutting is no longer available. It is recommended to perform cutting in the original framework instead.
> Below are examples of input and output cutting for TensorFlow protobuf, TensorFlow SavedModel, and ONNX formats in legacy conversion API, including examples of how to perform identical cut in the original framework.
> For PyTorch, TensorFlow 2 Keras, and PaddlePaddle, it is recommended to change the original model code to perform the model cut. 

Reviewer: ```suggestion
Performing surgery by cutting model inputs and outputs from a model is no longer available in the new conversion API. Instead, we recommend performing the cut in the original framework. 
Below are examples of model cutting of TensorFlow protobuf, TensorFlow SavedModel, and ONNX formats with the legacy conversion API, compared to achieving the same cut with tools provided by the Tensorflow and ONNX frameworks.
For PyTorch, TensorFlow 2 Keras, and PaddlePaddle, we recommend changing the original model code to perform the model cut.
```

``TensorFlow protobuf format / tf.Graph / tf.GraphDef``
#######################################################

.. list-table::

>    :header-rows: 1

Reviewer: I think it would be better to have a top-bottom layout of the comparison, as opposed to left-right columns. The code doesn't really align anyways.

Author: Done.

            - Not available in OVC tool. Please check Python API.

Cutting Off Parts of a Model
############################

Performing surgery by cutting model inputs and outputs from a model is no longer available in the new conversion API. Instead, we recommend performing the cut in the original framework. 
Below are examples of model cutting of TensorFlow protobuf, TensorFlow SavedModel, and ONNX formats with the legacy conversion API, compared to achieving the same cut with tools provided by the Tensorflow and ONNX frameworks.
For PyTorch, TensorFlow 2 Keras, and PaddlePaddle, we recommend changing the original model code to perform the model cut.

``PyTorch``
###########

Model cut for PyTorch is not available in legacy API.  

> Example of input and output layer cut in original FW.

Reviewer 2: ```suggestion
When it is needed to remove a whole module from the model it is possible to replace such modules with `Identity`. Below is the example of removing `conv1` and `bn1` modules at the input and `fc` module at the output of the resnet50 model.
```

.. code-block:: py
   :force:

   import openvino as ov
   import torch
   import torchvision
   from torch.nn import Identity
     
   # Load pretrained model
   model = torchvision.models.resnet50(weights='DEFAULT')
     
   # input cut
   model.conv1 = Identity()
   model.bn1 = Identity()
     
   # output cut
   model.fc = Identity()
     
   # convert and compile the model
   ov_model = ov.convert_model(model, input=([-1,64,-1,-1], torch.float32))
   compiled_model = ov.compile_model(ov_model)

> Example of cutting model outputs.

Reviewer 2: ```suggestion
When it is needed to remove one or more outputs from the model it is possible to create a wrapper for the model and only output the needed output. Below is the example of removing second output from the model.
```

Below are examples of model cutting of TensorFlow protobuf, TensorFlow SavedModel, and ONNX formats with the legacy conversion API, compared to achieving the same cut with tools provided by the Tensorflow and ONNX frameworks.
For PyTorch, TensorFlow 2 Keras, and PaddlePaddle, we recommend changing the original model code to perform the model cut.

> Note: This guide does not cover the cutting by input port of an operation. This type of cut is available only in legacy API. 

Reviewer: Honestly, I don't know what that means. Do we expect most readers to know what we're talking about in this note? If not, then it's mostly distracting IMO.

Reviewer 3: MO provides capability to cut a model by input port of an operation, for example, `--input 1:name_op` or `--output 0:name_op` but this guide does not tell how to handle such case. So we should explicitly tell an user that this case is not covered.
If the current sentence is unclear, let us clarify this case a bit more.

```
This guide does not cover the cutting a model by input port of an operation that MO tool provides using `input` and `output` options, for example, `--input 1:name_op`. 
```

Author: Corrected note according to Roman suggestion.

Reviewer: OK, that's more clear. Thank you. Ship it!
